navigation/src/main/java/com/graphhopper/navigation/NavigateResponseConverter.java

```diff
@@ -158,6 +159,54 @@ private static void putLegInformation(ObjectNode legJson, ResponsePath path, int
         legJson.put("distance", Helper.round(distance, 1));
     }
 
+
+    /**
+     * fix the first IntersectionDetail.
+     *
+     * The first Intersection of the first step should only have 1 bearing and one "out" entry
+     *
+     */
+    private static List<PathDetail> fixFirstIntersectionDetail( List<PathDetail> intersectionDetails) {
+        List<PathDetail> list = new ArrayList<>();
+
+        // job1: find out the interesting part of the intersectionDetails
+        if (intersectionDetails.size() < 2) {
+            // LOG Error? User didnt ask for intersections ?
+            return intersectionDetails;
+        }
+
+        Map<String, Object> newFirstIntersection = new HashMap<>();
+
+        final Map<String, Object> firstItersectionMap = (Map<String, Object>) intersectionDetails.get(0).getValue();
+
+        int out = (int) firstItersectionMap.get("out");
+        newFirstIntersection.put( "out", 0); // we will have only one entry
+
+        // bearings
+        List<Integer> firstBearingsList = (List<Integer>) firstItersectionMap.get("bearings");
+        List<Integer> firstBearings = new ArrayList<>();
+        firstBearings.add(firstBearingsList.get(out));
```

Comment:
nit: the naming of firstBearings and firstBearingsList feels odd. To me this feels like an error could creep in too easily by mixing up the two. (same for entries)